faiss/python/swigfaiss.swig

```diff
@@ -1022,17 +1022,17 @@ PyObject *swig_ptr (PyObject *a)
         return SWIG_NewPointerObj(data, SWIGTYPE_p_bool, 0);
     }
     if(PyArray_TYPE(ao) == NPY_UINT64) {
-#ifdef SWIGWORDSIZE64
-        return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long, 0);
-#else
+#if (__WORDSIZE == 32)
         return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long_long, 0);
+#else
+        return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long, 0);
 #endif
     }
     if(PyArray_TYPE(ao) == NPY_INT64) {
-#ifdef SWIGWORDSIZE64
-        return SWIG_NewPointerObj(data, SWIGTYPE_p_long, 0);
+#if  (__WORDSIZE == 32)
+        return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long_long, 0);
```

Comment:
please check on `__SIZEOF_LONG__` and generate `SWIGTYPE_p_long_long` or `SWIGTYPE_p_long`, not unsigned types